packages/common/http/src/client.ts

```diff
@@ -568,23 +571,32 @@ export class HttpClient {
             return res$.pipe(map((res: HttpResponse<any>) => {
               // Validate that the body is an ArrayBuffer.
               if (res.body !== null && !(res.body instanceof ArrayBuffer)) {
-                throw new Error('Response is not an ArrayBuffer.');
+                const errorMessage = (typeof ngDevMode === 'undefined' || ngDevMode) ?
+                    'Response is not an ArrayBuffer.' :
+                    '';
+                throw new RuntimeError(RuntimeErrorCode.INVALID_HTTP_RESPONSE, errorMessage);
```

Comment:
While the code is not fully aligned in the repository, here is a pattern that we've started to use recently:
```suggestion
                // Somewhere at the top of the file (after all imports)
                const NG_DEV_MODE = typeof ngDevMode === 'undefined' || ngDevMode;

                // At the actual call site:
                throw new RuntimeError(RuntimeErrorCode.INVALID_HTTP_RESPONSE, NG_DEV_MODE && 'Response is not an ArrayBuffer.');
```
It retains the tree-shaking, but it's less verbose. Could you please update the code to use that pattern?